include/c3d/base.h

```diff
@@ -1,41 +1,148 @@
+/**
+ * @file base.h
+ * @brief Base citro3d functions
+ */
 #pragma once
 #include "buffers.h"
 #include "maths.h"
 
+/// Default command buffer size
 #define C3D_DEFAULT_CMDBUF_SIZE 0x40000
 
+/**
+ * @brief Data type of indicies for use with \ref C3D_DarawElements.
+ */
 enum
 {
-	C3D_UNSIGNED_BYTE = 0,
-	C3D_UNSIGNED_SHORT = 1,
+	C3D_UNSIGNED_BYTE = 0,  ///< Unsigned 8-bit interger
+	C3D_UNSIGNED_SHORT = 1, ///< Unsigned 16-bit interger
 };
 
+/**
+ * @brief Initalizes citro3d
+ * @param[in] cmdBufSize Desired size of GPU command buffer. \ref C3D_DEFAULT_CMDBUF_SIZE should generally be used, unless there are given memory consumption or scene complexity requirements.
+ * @return true if library was initalized sucessfuly, false if there was an error.
+ */
 bool C3D_Init(size_t cmdBufSize);
+
+/**
+ * @brief De-initalizes citro3d
+ * @sa C3D_Init()
+ */
 void C3D_Fini(void);
 
+/**
+ * @brief Gets command buffer usage
+ * @return Fraction of command buffer used.
+ */
 float C3D_GetCmdBufUsage(void);
 
+/**
+ * @brief Binds a shader program to the current rendering state
+ * @param[in] program Specifies the pointer to a shader program object whose executables are to be used
+ *       as part of current rendering state.
+ */
 void C3D_BindProgram(shaderProgram_s* program);
 
+/**
+ * @brief Sets the viewport for the current framebuffer
+ * @note This value is modified by \ref C3D_FrameDrawOn(). (which is set using \ref C3D_RenderTargetSetOutput())
+ * @param[in] x X offset from the top left corner of the viewport in pixels.
```

Comment:
Sidenote. I suspect this might be the bottom left corner instead like in GL (and with Y pointing up); can't remember/haven't tested though. Unrelated to 3DS screen orientation, i.e. it would be bottom *right* corner for the screens.

Reply:
I'll test this later. Sorry about that.

Reply:
It appears to be the bottom left corner of the screen when the 3ds fliped with the touch screen facing right and the other screen facing left.
`C3D_SetViewport(40, 0, 240, 400);` moves the triangle to the top of the screen.
`C3D_SetViewport(40, 40, 240, 400);` moves the triangle to the top of the screen and a little bit left.

Reply:
Oh, I misread. GL uses the bottom left corner for this. I changed it to top left corner because I suspected this would be the case. Half-assed testing seems to show this is the case, but with as you said, the Y axis being along the width of the screen and the X axis being along the height of the screen.